### DISCUSSION TITLE
Writing CAM awareness data to the output.csv

### DISCUSSION THREAD

**Discussion by SafrasIqbal** [eclipse/mosaic#167] [Q&A] [answered]
Hello There,
I am using **VehicleCamSendingApp** class to send CAM messages. And looking to record CAM data from each vehicle in the output.csv file. I was trying to expend the "V2xMessageTransmission" subscription in output_config.xml by adding vehicle awareness data (Speed, Heading ..etc). However, it didn't work for me. 

Please advise me the best approach that I can follow to implement  and record CAM data. Thank you.

**Comment by schwepmo** [comment, marked as answer]
I don't think this is directly possible as the `V2xMessageTransmission` only holds the abstract `V2xMessage` class (not limited to `Cam`s).
The easiest solutions I can think of is to either use the `VehicleUpdates` subscription, as it seems to hold all the data you're interested in,
or write a small application, that receives and logs all `Cam`s manually.

**Reply by SafrasIqbal** [reply]
Hi @schwepmo, Thank you for your response. Understood, for now I am using VehicleUpdates. Thank you.
